src/plugins/intel_npu/src/plugin/npuw/attention.hpp

```diff
@@ -0,0 +1,185 @@
+    // as it reflects the attention input here.
+    PPtr _mask;
+    ov::Shape _mask_shape;
+
+    std::size_t query_len() const {
+        // Put the mask's innermost dimension dynamic
+        NPUW_ASSERT(_mask_shape.size() == 4);
+        return _mask_shape.at(2);
+    }
+
+    std::size_t context_len() const {
+        // Put the mask's innermost dimension dynamic
+        NPUW_ASSERT(_mask_shape.size() == 4);
+        return _mask_shape.at(3);
+    }
+};
+
+}  // namespace function
+
+namespace compiled {
+
+namespace {
+template <typename T>
+void prepare_mask(ov::SoPtr<ov::ITensor> tensor) {
+    NPUW_ASSERT(tensor->is_continuous());
+    ov::npuw::util::fill_tensor<T>(tensor, 0);
+
+    // Initialize the top-right triangle with -inf
+    const auto shape = tensor->get_shape();
+    NPUW_ASSERT(shape.size() == 4);
+    NPUW_ASSERT(shape[0] == 1);
+    NPUW_ASSERT(shape[1] == 1);
+
+    const auto q_size = shape[2];
+    const auto ctx_size = shape[3];
+    for (std::size_t i = 0; i < q_size - 1; i++) {
+        T* pRow = tensor->data<T>() + i*ctx_size;
+        for (std::size_t j = ctx_size - q_size + i; j < ctx_size; j++) {
+            pRow[j] = std::numeric_limits<T>::lowest();
+        }
+    }
+
+    // for a matrix like below where i = -inf (of type T)
+    // <-past -> < q >
+    // 000000000 0iiii
+    // 000000000 00iii
+    // 000000000 000ii
+    // 000000000 0000i
+    // 000000000 00000
+}
+} // anonymous namespace
+
+// Compile-time attention information. Not much different from the above
+struct Attention {
+    std::size_t query_size = 0u;
+    std::size_t context_size = 0u;
+
+    struct Param {
+        std::size_t idx;  // function input index for this spatial parameter
+        std::size_t dim;
+    };
+    std::vector<Param> params;
+    std::size_t mask_idx = 0u;
+
+    ov::Tensor attend_all;
+
+    Attention() = default;
+    Attention(const function::Attention& d, const std::shared_ptr<ov::Model>& m)
+        : query_size(d.query_len())
+        , context_size(d.context_len()) {
+        for (auto&& input : d._inputs) {
+            std::size_t p_idx = m->get_parameter_index(input.param);
+            params.push_back(Param{p_idx, input.dim});
+        }
+        mask_idx = m->get_parameter_index(d._mask);
+
+        // Create a mask data tensor here. Technically it is a runtime parameter,
+        // but in fact it doesn't change in runtime (we ignore what user passes us!)
+        // FIXME: Probably it is wrong
+        const auto mask_type = d._mask->get_element_type();
+        attend_all = ov::Tensor(mask_type, d._mask_shape);
+
+        switch (mask_type) {
+        case ov::element::f16:
+            prepare_mask<ov::float16>(ov::get_tensor_impl(attend_all));
+            break;
+        case ov::element::f32:
+            prepare_mask<float>(ov::get_tensor_impl(attend_all));
+            break;
+        default:
+            OPENVINO_THROW("Dynamic attenion mask type ", mask_type, " is not supported yet");
```

Comment:
In fact, we're using the graph_mask calculated in the graph itself. This `attent_all` is used for generate stage only where it is not actually needed (we can get rid of the mask argument at all)